internal/controllers/module_reconciler.go

```diff
@@ -315,6 +333,37 @@ func (mrh *moduleReconcilerHelper) prepareSchedulingData(ctx context.Context,
 	return result, errs
 }
 
+func (mrh *moduleReconcilerHelper) prepareImages(ctx context.Context, mod *kmmv1beta1.Module,
+	targetedNodes []v1.Node) ([]kmmv1beta1.ModuleImageSpec, error) {
+
+	var (
+		res    []kmmv1beta1.ModuleImageSpec
+		errs   []error
+		logger = log.FromContext(ctx)
+	)
+	for _, node := range targetedNodes {
+		kernelVersion := strings.TrimSuffix(node.Status.NodeInfo.KernelVersion, "+")
+		mld, err := mrh.kernelAPI.GetModuleLoaderDataForKernel(mod, kernelVersion)
```

Comment:
This flow is also used in prepareSchedulingData function. 
WDYT about getting the MLDs beforehand and then passing them as argument both to prepareImages and to prepareSchedulingData?  It might also require some refactoring of the code

Reply:
We disagreed (:)) to leave it as is for now. I will try to insert `TrimSuffix` inside `GetModuleLoaderDataForKernel` in a followup PR.